tests/FakeItEasy.Specs/ArgumentCapturingSpecs.cs

```diff
@@ -0,0 +1,339 @@
+
+            "And a method on the fake is called"
+                .x(() => fake.Invoke(-354897, 3));
+
+            "When we assert that the call must've happened with a constrained capturing argument and another non-matching argument"
+                .x(() => exception = Record.Exception(
+                    () => A.CallTo(() =>
+                            fake.Invoke(An<int>.That.IsCapturedTo(capturedArgument).And.IsGreaterThan(0), -3))
+                        .MustHaveHappened()));
+
+            "Then an exception is thrown, indicating the constraint on the argument"
+                .x(() => exception.Should().BeAnExceptionOfType<ExpectationException>()
+                    .WithMessageModuloLineEndings(@"
+
+  Assertion failed for the following call:
+    System.Action`2[System.Int32,System.Int32].Invoke(arg1: <greater than 0>, arg2: -3)
+  Expected to find it once or more but didn't find it among the calls:
+    1: System.Action`2[System.Int32,System.Int32].Invoke(arg1: -354897, arg2: 3)
+
+"));
+        }
+
+        [Scenario]
+        public static void MismatchedVerifyWithNegatedConstrainedCapture(
+            Action<int, int> fake, CapturedArgument<int> capturedArgument, Exception exception)
+        {
+            "Given a fake"
+                .x(() => fake = A.Fake<Action<int, int>>());
+
+            "And a capturedArgument instance"
+                .x(() => capturedArgument = new CapturedArgument<int>());
+
+            "And a method on the fake is called"
+                .x(() => fake.Invoke(-354897, 3));
+
+            "When we assert that the call must've happened with a negated constrained capturing argument and another non-matching argument"
+                .x(() => exception = Record.Exception(
+                    () => A.CallTo(() =>
+                            fake.Invoke(An<int>.That.IsCapturedTo(capturedArgument).And.Not.IsGreaterThan(0), -3))
+                        .MustHaveHappened()));
+
+            "Then an exception is thrown, indicating the constraint on the argument"
+                .x(() => exception.Should().BeAnExceptionOfType<ExpectationException>()
+                    .WithMessageModuloLineEndings(@"
+
+  Assertion failed for the following call:
+    System.Action`2[System.Int32,System.Int32].Invoke(arg1: <not greater than 0>, arg2: -3)
+  Expected to find it once or more but didn't find it among the calls:
+    1: System.Action`2[System.Int32,System.Int32].Invoke(arg1: -354897, arg2: 3)
+
+"));
+        }
+
+        [Scenario]
+        public static void CaptureMutatingArgument(
+            Action<IList<int>> fake, IList<int> listOfInts, CapturedArgument<IList<int>> capturedArgument)
+        {
+            "Given a fake"
+                .x(() => fake = A.Fake<Action<IList<int>>>());
+
+            "And a capturedArgument instance"
+                .x(() => capturedArgument = new CapturedArgument<IList<int>>(l => new List<int>(l!)));

```

Comment:
Not that it changes anything, but why `new List<int>(l)` instead of `l.ToList()`?